pkg/tests/apis/playlist/playlist_test.go

```diff
@@ -14,21 +14,50 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
 	"github.com/grafana/grafana/pkg/services/playlist"
 	"github.com/grafana/grafana/pkg/tests/apis"
+	"github.com/grafana/grafana/pkg/tests/testinfra"
 )
 
 func TestPlaylist(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping integration test")
 	}
-	helper := apis.NewK8sTestHelper(t)
+
+	t.Run("default setup", func(t *testing.T) {
+		doPlaylistTests(t, apis.NewK8sTestHelper(t, testinfra.GrafanaOpts{
+			AppModeProduction: true, // do not start extra port 6443
+			DisableAnonymous:  true,
+			EnableFeatureToggles: []string{
+				featuremgmt.FlagGrafanaAPIServer,
+			},
+		}))
+	})
+
+	t.Run("with k8s api flag", func(t *testing.T) {
```

Comment:
^^ this is the important part of this PR.  It now runs the same full test suite with the flag enabled and without.  We can soon do the same with the dual write storage path https://github.com/grafana/grafana/pull/77604